pkg/provider/azure_controller_common.go

```diff
@@ -529,71 +560,64 @@ func (c *controllerCommon) GetDiskLun(diskName, diskURI string, nodeName types.N
 	return -1, provisioningState, fmt.Errorf("%s for disk %s", consts.CannotFindDiskLUN, diskName)
 }
 
-// SetDiskLun find unused luns and allocate lun for every disk in disksPendingAttach map.
-// Return err if not enough luns are found.
-func (c *controllerCommon) SetDiskLun(nodeName types.NodeName, disksPendingAttach map[string]*AttachDiskOptions) error {
+// SetDiskLun finds an unused lun and allocates lun for disk pending attach.
+// Return err if not available lun cannot be found.
+func (c *controllerCommon) SetDiskLun(nodeName types.NodeName, diskPendingAttach *AttachDiskParams) error {
+	uri := diskPendingAttach.diskURI
```

Comment:
np: wouldn't hurt to check if `diskPendingAttach == nil` although given where it is called, it would be impossible for it to be `nil`